File: CS/NavigationFrameWizard/Customer.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NavigationFrameWizard {
    public class Customer : INotifyPropertyChanged {

        
        protected string _FirstName;

        public string FirstName {
            get {
                return this._FirstName;
            }

            set {
                if(this._FirstName != value) {
                    this._FirstName = value;
                    this.OnPropertyChanged("FirstName");
                }
            }
        }

        
        protected string _SecondName;

        public string SecondName {
            get {
                return this._SecondName;
            }

            set {
                if(this._SecondName != value) {
                    this._SecondName = value;
                    this.OnPropertyChanged("SecondName");
                }
            }
        }

        
        protected DateTime _Birthday;

        public DateTime Birthday {
            get {
                return this._Birthday;
            }

            set {
                if(this._Birthday != value) {
                    this._Birthday = value;
                    this.OnPropertyChanged("Birthday");
                }
            }
        }

        protected ObservableCollection<Order> _Orders;

        public ObservableCollection<Order> Orders {
            get {
                if(this._Orders == null) {
                    this._Orders = new ObservableCollection<Order>();
                }

                return this._Orders;
            }
        }

        public void OnPropertyChanged(string info) {
            if(this.PropertyChanged != null) {
                this.PropertyChanged(this, new PropertyChangedEventArgs(info));
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;
    }
}

File: VB/NavigationFrameWizard/Customer.vb
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace NavigationFrameWizard
    Public Class Customer
        Implements INotifyPropertyChanged


        Protected _FirstName As String

        Public Property FirstName() As String
            Get
                Return Me._FirstName
            End Get

            Set(ByVal value As String)
                If Me._FirstName <> value Then
                    Me._FirstName = value
                    Me.OnPropertyChanged("FirstName")
                End If
            End Set
        End Property


        Protected _SecondName As String

        Public Property SecondName() As String
            Get
                Return Me._SecondName
            End Get

            Set(ByVal value As String)
                If Me._SecondName <> value Then
                    Me._SecondName = value
                    Me.OnPropertyChanged("SecondName")
                End If
            End Set
        End Property


        Protected _Birthday As Date

        Public Property Birthday() As Date
            Get
                Return Me._Birthday
            End Get

            Set(ByVal value As Date)
                If Me._Birthday <> value Then
                    Me._Birthday = value
                    Me.OnPropertyChanged("Birthday")
                End If
            End Set
        End Property

        Protected _Orders As ObservableCollection(Of Order)

        Public ReadOnly Property Orders() As ObservableCollection(Of Order)
            Get
                If Me._Orders Is Nothing Then
                    Me._Orders = New ObservableCollection(Of Order)()
                End If

                Return Me._Orders
            End Get
        End Property

        Public Sub OnPropertyChanged(ByVal info As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(info))
        End Sub

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
    End Class
End Namespace
